jani_generator/src/jani_generator/scxml_helpers/scxml_event_processor.py

```diff
@@ -60,13 +66,23 @@ def implement_scxml_events_as_jani_syncs(
         event_automaton = JaniAutomaton()
         event_automaton.set_name(event_name)
         event_automaton.add_location("waiting", is_initial=True)
-        event_action_names.append(event_name_on_send)
+        if add_timer_syncs:
+            # Additional self-loop in the waiting state, allowing the global timer to tick
+            # only if all events have been processed
+            event_automaton.add_edge(JaniEdge({
+                "location": "waiting",
+                "destinations": [{
+                    "location": "waiting",
+                    "probability": {"exp": 1.0},
+                    "assignments": []
+                }],
+                "action": "global_timer_enable"
+            }))
```

Comment:
why do we need this?

Reply:
This is used to ensure we do not have any unprocessed event (that means, there are no event-automata in the "received" state) before advancing the global timer / processing the timer callbacks. In a way, we can say it gives priority to events over timers